numba/np/npyfuncs.py

```diff
@@ -243,9 +243,35 @@ def np_int_udivrem_impl(context, builder, sig, args):
     return context.make_tuple(builder, sig.return_type, [div, rem])
 
 
-# implementation of int_fmod is in fact the same as the unsigned remainder,
-# that is: srem with a special case returning 0 when the denominator is 0.
-np_int_fmod_impl = np_int_urem_impl
+def np_int_fmod_impl(context, builder, sig, args):
+    # implementation will be very similar to np_int_urem_impl, but
+    # the output sign will match the sign of the numerator
+    _check_arity_and_homogeneity(sig, args, 2)
+
+    num, den = args
+    ty = sig.args[0]  # any arg type will do, homogeneous
+
+    ZERO = context.get_constant(ty, 0)
+    den_not_zero = builder.icmp_signed("!=", ZERO, den)
+    bb_no_if = builder.basic_block
+    with cgutils.if_unlikely(builder, den_not_zero):
+        bb_if = builder.basic_block
+
+        # use the abs of the numerator and denominator
+        num_neg = builder.icmp_signed("<", num, ZERO)
+        den_neg = builder.icmp_signed("<", den, ZERO)
+        num_abs = builder.select(num_neg, builder.neg(num), num)
+        den_abs = builder.select(den_neg, builder.neg(den), den)
+        mod = builder.urem(num_abs, den_abs)
```

Comment:
Rather than checking if the arguments are negative, can't you just use [`llvm.srem`](https://llvm.org/docs/LangRef.html#srem-instruction) instead?

Reply:
Good catch! I had previously misread the docs for `srem` and had thought it used the sign of the denominator. That'll simplify the code a bunch